### ISSUE TITLE
libsolv isn't filtering kernel flavors correctly with urpmreorder enabled

### ISSUE THREAD

**Issue by Conan-Kudo** [openSUSE/libsolv#433]
I got a bug report ([mga#28272](https://bugs.mageia.org/show_bug.cgi?id=28272)) about `dnf system-upgrade` doing the wrong thing for upgrades from Mageia 7 to Mageia 8, where `kernel-server` gets installed even though `kernel-desktop` already exists and has an upgrade candidate.

This is with DNF 4.2.6 and libsolv 0.7.4, though I can reproduce it with the latest versions.

You can see what's happening with [the debugsolver data](https://ngompa.fedorapeople.org/mageia/debugdata-mga7-to-mga8-distrosync-20210203.tar).

**Comment by Conan-Kudo** [comment]
@mlschroe Can you please take a look at this, it's a pretty serious issue for Mageia 7->Mageia 8 upgrades....

**Comment by mlschroe** [comment]
I don't have much spare time for libsolv atm. Anyway, I'll try to have an answer till next week.

**Comment by mlschroe** [comment]
This is caused by the "yum style obsoletes" handling. The system has "crda" installed, and both kernel-server and kernel-desktop obsolete "crda". So it's handled like a package split.
(I'm happy that SUSE does not use this feature ;) )

**Comment by Conan-Kudo** [comment]
Couldn't the urpmreorder logic also make it so kernel flavors influence this? That is, the normal yumobsoletes behavior making it get considered as a package split doesn't get triggered?

**Comment by mlschroe** [comment]
No, the "rpmreorder" logic just orders solvables when there is a solvable to be chosen from a set. The yum style obsoletes handling is in the rule creation.

Just get rid of those obsoletes, I don't think they are correct as the kernel packages are not update candidates for those packages.

**Comment by Conan-Kudo** [comment]
The kernel maintainer believes it to be correct since the crda functionality moved into the kernel itself, though I agree with you that it doesn't make sense...
